addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -5451,6 +5454,25 @@ registry.ImageOptimize = ImageHandlerOption.extend({
             img.dataset.mimetype = 'image/svg+xml';
         }
     },
+    /**
+     * @private
+     */
+    async _reapplyCurrentShape() {
+        const img = this._getImg();
+        if (img.dataset.shape) {
+            await this._loadShape(img.dataset.shape);
+            await this._applyShapeAndColors(true, (img.dataset.shapeColors && img.dataset.shapeColors.split(';')));
+        }
+    },
+    /**
+     * @override
+     */
+    _optionAllowedOnAllImages(node) {
+        return [
+            'we-row[string="Transform"]',
+            'we-button-group[string="Width"]',
+        ].some(selector => node.matches && node.matches(selector));
+    },
```

Comment:
@xO-Tx I don't understand what this is? Can you explain? In any case, this is worth comments but I don't think this is necessary anymore. Now that the Image options are regrouped, either the UI is declared inside or not.

Also, you don't need an array, you can match multiple selectors at once with `node.matches('a, b')`

Reply:
1- Actually, after the merge of `ImageTools` & `ImageOptimize`, the code on `_renderCustomXML()` will remove all nodes from `uiFragment` if image is not supported (`_isImageSupportedForProcessing(img) === false`), so we need this to exclude the "crop" & "transform" since they are included now in the option's template.

2- Yes it's simpler with multiple selectors, thanks for reminding me about it :+1:

Reply:
Mmh ok indeed. Although I think that code removing all nodes is historic (from the time we had error message we needed to remove etc, not entire widget). I think using the `_computeWidgetVisibility` system would be more natural in this case as you only want to show/hide widget according to the image. Don't you agree?

Reply:
Yes indeed :+1:, I tough it was something for `BackgroundOptimize` so I avoided changing it, but as you said, I think it just removes nodes from `uiFragment` instead of hiding them now. So I updated the code to use an `_isImageProcessingWidget()` method to check if the widget should be shown on supported images only _(this method will be overridden by ImageTools to include shapes)_